packages/next-app/src/app/stableswapsimulator/(components)/TokenForm.tsx

```diff
@@ -0,0 +1,125 @@
+import { getStableSwapSimulatorTokensSchema } from "@balancer-pool-metadata/schema";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as Dialog from "@radix-ui/react-dialog";
+import { useForm } from "react-hook-form";
+
+import Button from "#/components/Button";
+import { Input } from "#/components/Input";
+import { TokensData, useStableSwap } from "#/contexts/StableSwapContext";
+
+export default function TokenForm({
+  symbolToEdit,
+  close,
+}: {
+  symbolToEdit?: string;
+  close?: () => void;
+}) {
+  const { initialData, setInitialData } = useStableSwap();
+  const stableSwapTokensSchema = getStableSwapSimulatorTokensSchema({
+    symbolToEdit,
+    existentSymbols: initialData?.tokens?.map((token: TokensData) => {
+      return token.symbol;
+    }),
+  });
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<typeof stableSwapTokensSchema._type>({
+    resolver: zodResolver(stableSwapTokensSchema),
+  });
+
+  const currentToken =
+    initialData?.tokens?.find(
+      (token: TokensData) => token.symbol == symbolToEdit
+    ) || undefined;
+
+  const validateUniqueSymbol = (symbol: string) => {
+    const tokensToCheck = initialData?.tokens?.filter(
+      (token: TokensData) => token.symbol != symbolToEdit
+    );
+    const symbolsToCheck = tokensToCheck?.map(
+      (token: TokensData) => token.symbol
+    );
+    if (symbolsToCheck?.includes(symbol)) {
+      return "This symbol already exists. Please define another name.";
+    }
+    return true;
+  };
+
+  function addOrEditToken(data: TokensData) {
+    if (symbolToEdit) {
+      setInitialData({
+        ...initialData,
+        tokens: initialData?.tokens?.map((token: TokensData) => {
+          if (token.symbol == symbolToEdit) {
```

Comment:
since this would always be `false` when `symbolToEdit` is not present you could really just have this condition, no?
```suggestion
        tokens: (initialData?.tokens || [])?.map((token: TokensData) => {
          if (token.symbol == symbolToEdit) {
```
but when would `initialData.tokens` be undefined/null? Shouldn't we initialize it with an empty array at least?